### PR TITLE
Migrate away from `SimpleDispatchInfo`

### PR DESCRIPTION
**Pull request by kianenigma**
- Deprecates SimpleDispatchInfo, replaced with impls for `Weight`, `(Weight, Class)` and `(Weight, Class, PaysFee)`
- Revamps all the docs accordingly. Since the new weight system is being prepared and will be presented in Sub0, this is quite important, the old docs were very much out of date. 

I brought this up (intention to deprecate `SimpleDispatchInfo`) and there were no objections, so I assume it is okay. `SimpleDispatchInfo` was verbose, not flexible (i.e. for each permutation of `(weight, class, fee)` you need a new enum variant) and does not have any added value. I always saw it as hasty mistake which was added in the first version of weights.

So all in all, If this is going to happen, better for it to happen now before 2.0. 

This PR should change no logic except for in `weight.rs`. substitution regex: 

```
// FixedNormal pays fee, default class is Normal, only replace weight
SimpleDispatchInfo::FixedNormal\((.*)\)
$1

// FixedOperational pays fee, no need to set third value
SimpleDispatchInfo::FixedOperational\((.*)\)
\($1, frame_support::weights::DispatchClass::Operational\)

// FixedMandatory pays fee, no need to set third value
SimpleDispatchInfo::FixedMandatory\((.*)\)
\($1, frame_support::weights::DispatchClass::Mandatory\)
```

- [x] polkadot companion: https://github.com/paritytech/polkadot/pull/1008
- [x] clean some imports.

### CODE REVIEW

File: frame/executive/src/lib.rs [hunk 1]
```diff
@@ -416,22 +416,22 @@ mod tests {
...
 					// NOTE: does not make any different.
 					let _ = frame_system::ensure_signed(origin);
 				}
-				#[weight = SimpleDispatchInfo::FixedOperational(200)]
+				#[weight = (200, frame_support::weights::DispatchClass::Operational)]
 				fn some_root_operation(origin) {
 					let _ = frame_system::ensure_root(origin);
 				}
-				#[weight = SimpleDispatchInfo::InsecureFreeNormal]
+				#[weight = 0]
```
Comment (kianenigma) [inline frame/executive/src/lib.rs:434]: interesting case not automatically migrated.

File: frame/democracy/src/lib.rs [hunk 2]
```diff
@@ -1251,15 +1251,15 @@ decl_module! {
...
+		#[weight = MINIMUM_WEIGHT]
 		fn proxy_remove_vote(origin, index: ReferendumIndex) -> DispatchResult {
 			let who = ensure_signed(origin)?;
 			let target = Self::proxy(who).and_then(|a| a.as_active()).ok_or(Error::<T>::NotProxy)?;
 			Self::try_remove_vote(&target, index, UnvoteScope::Any)
 		}
 
 		/// Enact a proposal from a referendum. For now we just make the weight be the maximum.
-		#[weight = SimpleDispatchInfo::MaxNormal]
+		#[weight = Weight::max_value()]
```
Comment (kianenigma) [inline frame/democracy/src/lib.rs:1262]: Interesting case not handled by the regex.

File: frame/timestamp/src/lib.rs [hunk 3]
```diff
@@ -148,7 +148,7 @@ decl_module! {
 		/// `MinimumPeriod`.
 		///
 		/// The dispatch origin for this call must be `Inherent`.
-		#[weight = SimpleDispatchInfo::FixedMandatory(MINIMUM_WEIGHT)]
+		#[weight = (MINIMUM_WEIGHT, frame_support::weights::DispatchClass::Mandatory)]
```
Comment (shawntabrizi) [inline frame/timestamp/src/lib.rs:151]: Can you just import this above?

If it is too much of a pain, let me know, I am happy to do it :)

Comment (kianenigma) [reply frame/timestamp/src/lib.rs:151]: no problem, I already noted it above as a todo to clean this stuff.

**Comment by gnunicorn** [comment]
tests failing.

**Comment by kianenigma** [comment]
should be good now.  @gnunicorn

**Comment by kianenigma** [comment]
Updated the polkadot PR as well and both should be mergeable now.

**Pull request merged**